File: 0x08-python-more_classes/2-rectangle.py
#!/usr/bin/python3
"""Rectangle"""

class Rectangle:
    """class Rectange with private and public instances"""
    
    _Rectangle__height = None
    _Rectangle_width = None

    def __init__(self, widthVal=0, heightVal=0):
        self.height = heightVal
        self.width = widthVal

    @property
    def height(self):
        return self._Rectangle__height

    @height.setter
    def height(self, value):
        if not isinstance(value, int):
            raise TypeError("height must be an integer")
        if (value < 0):
            raise ValueError("height must be  >= 0")
        self._Rectangle__height = value

    @property
    def width(self):
        return self._Rectangle__width

    @width.setter
    def width(self, value):
        if not isinstance(value, int):
            raise TypeError("width must be an integer")
        if (value < 0):
            raise ValueError("width must be  >= 0")
        self._Rectangle__width = value

    def area(self):
        return (self.height * self.width)

    def perimeter(self):
        if (self.height == 0 or self.width == 0):
            return 0
        else:
            return (self.height + self.width) * 2

File: 0x13-javascript_objects_scopes_closures/2-rectangle.js
#!/usr/bin/node

/**
 * a class Rectangle that defines a rectangle:
 * If w or h is equal to 0 or not a positive integer, create an empty object
 */

class Rectangle {
  constructor (w, h) {
    if (w > 0 && h > 0) {
      this.width = w;
      this.height = h;
    }
  }
}

// const r1 = new Rectangle(2, 3);
// console.log(r1);
// console.log(r1.width);
// console.log(r1.height);

// const r2 = new Rectangle(2, -3);
// console.log(r2);
// console.log(r2.width);
// console.log(r2.height);

// const r3 = new Rectangle(2);
// console.log(r3);
// console.log(r3.width);
// console.log(r3.height);

// const r4 = new Rectangle(2, 0);
// console.log(r4);
// console.log(r4.width);
// console.log(r4.height);

module.exports = Rectangle;
